### ISSUE TITLE
The app is wrongly closed after entering and exiting immersive mode multiple times on different windows

### ISSUE THREAD

**Issue by Softvision-GeluHaiduc** [MozillaReality/FirefoxReality#1829]
**[Affected Versions]:**
- Firefox Reality 1.5-32552323 (4078203)

**[Affected Platforms]:**
- Oculus Go - Android 7.1.2
- Oculus Quest - Android 7.1.1
- HTC Vive Focus Plus - Android 7.1.2
- HTC Vive Focus - Android 7.1.2

**[Prerequisites]:**
- The headset is turned on and the browser is opened.

**[Steps to reproduce]:**
1. Open any video from the FxR homepage and enter immersive mode.
2. Exit immersive mode and click the “+” (New Window) button.
3. Repeat Steps 1 and 2 until there are three windows opened.
4. Cycle through the windows entering and exiting immersive mode in each one.
5. Repeat step 4 a few times and observe the behavior.

**[Expected results]:**
- Each video enters and exists immersive mode successfully.

**[Actual results]:**
- The app is closed.

**[Notes]:**
- The issue is consistently reproducible on all devices but the time until the app is exited differs.
- From the HTC Vive Focus Plus I’ve attached a screen recording of the issue [here](https://drive.google.com/file/d/11smHSia2DnTdbbn4tlJb7Kp5b_czaS_f/view?usp=sharing) and the logcat file [here](https://drive.google.com/file/d/1yJUzGfQoDSqbr-Z7OBqgeev_hyVaRayJ/view?usp=sharing). 
- From the Oculus Quest I’ve attached the logcat file [here](https://drive.google.com/file/d/1Fsba5BNJuB5GKEnqU2syMyWzvU28K6vA/view?usp=sharing).

**Comment by bluemarvin** [comment]
It looks like both linked logcat files are from Oculus. I've only been able to trigger the `RetainedDisplayListBuilder::MergeDisplay()` crash so far in my attempts to reproduce on a Quest. In the oculus logcat I see a crash but FxR not restarting is a little worrisome. I will try on HTC to see if I get different results.

**Comment by bluemarvin** [comment]
I haven't been able to reproduce this on HTC either yet.

**Comment by Softvision-GeluHaiduc** [comment]
@bluemarvin I’ve reattached the logcat files as a zip.
I'm not sure what went wrong the first time. Thanks!
[Logcat files.zip](https://github.com/MozillaReality/FirefoxReality/files/3615934/Logcat.files.zip)

**Comment by bluemarvin** [comment]
From the logs, it looks like a crash but the crash reporter is not able to restart FxR so the crash report is getting lost. With out being able to reproduce the crash, I'm going to guess this is the `RetainedDisplayListBuilder::MergeDisplay()` crash. It could also be a crash in the media process but I can't find anything specific in either log other than a crash has occurred. I'm going to look into fixing the crash reporter so that we look for crash files when FxR is restarted so we don't loose crashes like this.

**Comment by bluemarvin** [comment]
I created a patch that should help prevent the loss of crash reports: https://github.com/MozillaReality/FirefoxReality/pull/1847
Once it lands, if you could try and reproduce that would be appreciated.

**Comment by Softvision-GeluHaiduc** [comment]
@bluemarvin I saw that #1847 has landed.
Using the latest Firefox Reality 1.5-32601659 (0f16dc02) installed on the HTC Vive Focus Plus I managed to reproduce the issue, this time after the app was wrongly closed I opened the app again and the crash popup appeared.
The crash report can be found [here](https://crash-stats.mozilla.com/report/index/5387042c-a021-440c-8d89-5f9b50190918).
I also attached the logcat file. [log_Wed09-18-2019_9-15-48.36(vive).txt](https://github.com/MozillaReality/FirefoxReality/files/3624811/log_Wed09-18-2019_9-15-48.36.vive.txt)
